src/main/java/seedu/weme/ui/CommandBox.java

```diff
@@ -65,74 +66,138 @@ public CommandBox(CommandExecutor commandExecutor,
     private void handleKeyPress(KeyEvent keyEvent) {
         switch (keyEvent.getCode()) {
         case UP:
-            // As up and down buttons will alter the position of the caret,
-            // consuming it causes the caret's position to remain unchanged
         case DOWN:
-            handleLikeByKeyPress(keyEvent);
-            break;
         case LEFT:
         case RIGHT:
-            handleIndexToggleByKeyPress(keyEvent);
+            handleArrowKeyPress(keyEvent);
             break;
         case TAB:
             handleTabPress(keyEvent);
             break;
         default:
-            // let JavaFx handle the keypress
+        }
+    }
+
+    /**
+     * Handles arrow key press events.
+     * Offloads the real handling to the appropriate handlers.
+     * @param event an arrow key press event
+     */
+    private void handleArrowKeyPress(KeyEvent event) {
+        assert Set.of(KeyCode.UP, KeyCode.DOWN, KeyCode.LEFT, KeyCode.RIGHT).contains(event.getCode());
+
+        // Do nothing if caret is not at the end of the text
+        if (commandTextField.getCaretPosition() != commandTextField.getText().length()) {
+            return;
+        }
+
+        if (handleLikeByKeyPress(event)
+            || handleIndexToggleByKeyPress(event)
+            || handleMovingTextUsingKeyPress(event)) {
+            commandTextField.positionCaret(commandTextField.getText().length());
+            event.consume();
         }
     }
 
     /**
      * Changes index in the command box with key press to enable fast liking.
+     * @return whether this event is handled
      */
-    private void handleIndexToggleByKeyPress(KeyEvent event) {
+    private boolean handleIndexToggleByKeyPress(KeyEvent event) {
+        // Only handle left and right key press
+        if (!(event.getCode().equals(KeyCode.LEFT) || event.getCode().equals(KeyCode.RIGHT))) {
+            return false;
+        }
+
         final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(commandTextField.getText().trim());
         // Check if the input is a valid command
         if (!matcher.matches()) {
-            return;
+            return false;
         }
 
         if (!(matcher.group(COMMAND_WORD).equals(MemeLikeCommand.COMMAND_WORD)
-                || matcher.group(COMMAND_WORD).equals(MemeDislikeCommand.COMMAND_WORD))) {
+            || matcher.group(COMMAND_WORD).equals(MemeDislikeCommand.COMMAND_WORD))) {
```

Comment:
@moziliar too
is there actually any way to check whether we are currently in the meme tab? Right not in any other tab, when it sees `like 1` or `dislike 1` and you press the left or right arrow keys, it will increment/decrement the index, even though it is not valid, so it feels like a bug. 
Pressing up or down is fine because it tries to execute, fails then does the default behaviour of moving the cursor

Reply:
The way I can think of is to let `CommandBox` take in `ModelContext` as well

Reply:
@jonchan51 @moziliar I added `ModelContext` to `CommandBox` in the latest commit. Does it look ok?